lib/tasks/journeys_report.rake

```diff
@@ -97,24 +49,35 @@ namespace :journeys do
           from: move.from_location.title,
           to: move.to_location&.title,
           events: move.move_events.default_order.map do |move_event|
-            {
-              timestamp: move_event.client_timestamp.strftime('%d/%m %H:%M:%S'),
-              event: move_event.event_name,
-              notes: move_event.notes,
-              from_location: move_event.from_location&.title,
-              to_location: move_event.to_location&.title,
-            }
-          end,
+                    {
+                      timestamp: move_event.client_timestamp.strftime('%d/%m %H:%M:%S'),
+                      event: move_event.event_name,
+                      notes: move_event.notes,
+                      from_location: move_event.from_location&.title,
+                      to_location: move_event.to_location&.title,
+                    }
+                  end,
           journeys: journeys,
         }
       end
       suppliers << {
         supplier: supplier.key,
         moves: moves,
       }
-    end
-    File.open('journeys_report.json', 'w') do |file|
-      file.write(JSON.pretty_generate({ data: suppliers }))
+
+      File.open(filename, 'w') do |file|
+        file.write(JSON.pretty_generate({ data: suppliers })) # maybe change to .to_json for production
+      end
+
+      puts "Exporting #{filename} to S3..."
+      bucket_name = ENV.fetch('S3_REPORTING_BUCKET_NAME')
+      if bucket_name.present?
+        s3 = Aws::S3::Resource.new
+        obj = s3.bucket(bucket_name).object(filename)
+        obj.upload_file(filename)
```

Comment:
Can't we just upload what's in memory, here? Rather than reading the filename again?

Reply:
Me and @martyn-w discussed that and felt that for now is preferable to have it this way as we can just get the local file without uploading it (If you don't define a target bucket, it will just skip that step)